packages/qdrant-loader/tests/unit/cli/test_serve_cmd.py

```
@@ -0,0 +1,50 @@
+    for key in list(os.environ):
+        if key.startswith("WEBHOOK_SECRET") or key == "WEBHOOK_ENABLE_COGNITO_JWT":
+            monkeypatch.delenv(key, raising=False)
+
+
+def test_serve_fails_closed_before_state_manager_init(monkeypatch):
+    """No webhook secret configured: serve must raise ClickException before
+    ever reaching StateManager.initialize() (i.e. before any DB engine, queue,
+    or uvicorn/asyncio resource is created)."""
+    monkeypatch.setattr(
+        "qdrant_loader.cli.config_loader.load_config_with_workspace",
+        lambda *a, **k: None,
```

Ruff (0.15.21)

[warning] Unused lambda argument: `a`

(ARG005)